doc/user/content/guides/node-js.md

```diff
@@ -0,0 +1,245 @@
+
+Read a stream of updates from a materialized view named `my_view` into Node.js:
+
+```js
+const { Client } = require('pg');
+
+async function main() {
+  const client = new Client('postgres://materialize@localhost:6875/materialize');
+  await client.connect();
+
+  await client.query('BEGIN');
+  await client.query('DECLARE c CURSOR FOR TAIL my_view');
+
+  while (true) {
+    const res = await client.query('FETCH ALL c');
+    console.log(res.rows);
+  }
+}
+
+main();
+```
+
+The code above opens a long-lived transaction with `BEGIN` and uses [`TAIL` with `FETCH`](/sql/tail/#tailing-with-fetch) to receive a stream of updates to `my_view`. The [TAIL Output format](/sql/tail/#output) of `res.rows` is an array of view update objects.
+
+**Note on output format:** When a row of a tailed view is **updated,** two objects will show up in the `rows` array:
+
+```js
+[
+    ...
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '1',
+        my_column_one: 'ABC',
+        my_column_two: 'new_value'
+    },
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '-1',
+        my_column_one: 'ABC',
+        my_column_two: 'old_value'
+    },
+    ...
+]
+```
+
+An `mz_diff` value of `-1` indicates Materialize is deleting one row with the included values.  An update is just a deletion (`mz_diff: '-1'`) and an insertion (`mz_diff: '1'`) with the same `mz_timestamp`.
+
+<!--- NOT PUBLISHABLE UNTIL https://github.com/brianc/node-postgres/pull/2573 is merged
+### Using pg-query-stream
+
+```js
+const { Client } = require('pg');
+const QueryStream = require('pg-query-stream');
+
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+client.connect((err, client) => {
+  if (err) {
+    throw err;
+  }
+  const stream = client.query(new QueryStream('TAIL avg_bid', []));
+  stream.pipe(process.stdout);
+});
+```
+--->
+
+## Query
+
+Querying Materialize is identical to querying a traditional PostgreSQL database: Node.js executes the query, and Materialize returns the state of the view, source, or table at that point in time.
+
+Query a view `my_view` with a select statement:
+
+```js
+const { Client } = require('pg');
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+async function main() {
+  await client.connect();
+  const res = await client.query('SELECT * FROM my_view');
+  console.log(res.rows);
+};
+
+main();
+```
+
+Polling _(repeatedly querying)_ of a view is fine, and response times are much faster than traditional database queries because Materialize maintains materialized views in memory. For more, refer to the node-postgres documentation of [Queries in node-postgres](https://node-postgres.com/features/queries) and [pg.Result](https://node-postgres.com/api/result) for output reference.
```

Comment:
I think this should probably be moved to the section intro and rephrased as: 

> Because Materialize maintains materialized views in memory, response times are much faster than traditional database queries and polling (repeatedly querying) a view doesn't impact performance.

That way "For more" doesn't sound like you mean "For more information about materialized views". We can also abbreviate that to:

> For more details, see the [node-postgres query] and [pg.Result output] documentation.